client/src/schema/query.ts

```diff
@@ -192,28 +232,20 @@ export class Query<T extends XataRecord, R extends XataRecord = T> implements Ba
     return records[0] || null;
   }
 
-  async deleteAll(): Promise<number> {
+  /**async deleteAll(): Promise<number> {
     // TODO: Return number of affected rows
     return 0;
-  }
+  }**/
 
   async nextPage(size?: number, offset?: number): Promise<Page<T, R>> {
     return this.firstPage(size, offset);
   }
 
-  async previousPage(size?: number, offset?: number): Promise<Page<T, R>> {
-    return this.firstPage(size, offset);
-  }
-
   async firstPage(size?: number, offset?: number): Promise<Page<T, R>> {
     return this.getPaginated({ page: { size, offset } });
   }
 
   async lastPage(size?: number, offset?: number): Promise<Page<T, R>> {
     return this.getPaginated({ page: { size, offset, before: 'end' } });
   }
-
-  hasNextPage(): boolean {
-    return this.meta.page.more;
-  }
```

Comment:
Question. I see this method is being removed and `previousPage` as well. Can you explain why? If it's been moved I don't see where it is now.

Reply:
Yeah, those methods on a repository/query level did not make much sense. But its true that we were partially removing an easy way to do the following:

```ts
function *foo(page: Paginable) {
    yield page.records;
    if (page.hasNextPage()) return foo(page.nextPage());
}

foo(xata.db.users)
```

Added it back!